### DISCUSSION TITLE
Understanding the Database and extensibility

### DISCUSSION THREAD

**Discussion by mattchristenson** [NginxProxyManager/nginx-proxy-manager#2145] [Q&A]
tldr:  Can I change data in the database to reconfigure nginx?

Hi there!  I'm looking at potentially using NGINX Proxy Manager into our tooling, and I noticed in the setup that there is an opportunity to use a MySQL database instead of a sqllite file.  This got me thinking about extensibility.  Soon I'll want to be able to script changes in our CI/CD scripts for when we switch which environment is exposed (using a blue/green pattern).  If I adopt NPM, can I write update scripts to the database to change configuration, or is it better to go to the nginx config files and modify those?  If I do make a change in the database, can I do something that would trigger those new settings to go into effect?  I'm just curious what the NPM architecture looks like behind the scenes.
